policy_agent/policy_agent.py

```diff
@@ -0,0 +1,643 @@
+                            except ProfileAlreadyInEndpoint:
+                                _log.warning(
+                                    "Applying %s to Pod %s : Profile Already exists" % (profile, existing_pod.key))
+                        else:
+                            _log.warning("Pod %s is not yet processed" % (pod))
+
+                # declare Endpoints obj processed
+                self.epRV = ep.RV
+                self.endpoints[ep_key] = ep
+                del self.changed_endpoints[ep_key]
+
+
+class Resource():
+
+    """
+    Resource objects pull pertinent info from json blobs and maintain universal functions
+    """
+
+    def __init__(self, json):
+        """
+        On init, each Resource saves the raw json, pulls necessary info (unique),
+        and defines a unique key identifier
+        """
+        self.json = json
+        self.RV = json["metadata"]["resourceVersion"]
+        self.from_json(json)
+        self.key = self.get_key()
+
+    def from_json(self, json):
+        self.name = "noop"
+        return
+
+    def get_key(self):
+        return self.name
+
+    def __str__(self):
+        return "%s: %s\n%s" % (self.kind, self.key, self.json)
+
+
+class Namespace(Resource):
+
+    def from_json(self, json):
+        self.kind = "Namespace"
+        self.name = json["metadata"]["name"]
+
+        try:
+            self.policy = json["spec"]["experimentalNetworkPolicy"]
+        except KeyError:
+            _log.warning("Namespace does not have policy, assumed Open")
+            self.policy = "Open"
+
+    def get_key(self):
+        return self.name
+
+    def create_ns_profile(self):
+        # Derive NS tag
+        ns_tag = "namespace_%s" % self.name
+
+        # Add Tags
+        profile_path = PROFILE_PATH % {"profile_id": ns_tag}
+        _datastore_client.etcd_client.write(
+            profile_path + "tags", '["%s"]' % ns_tag)
+
+        # Determine rule set based on policy
+        default_allow = Rule(action="allow")
+
+        if self.policy == "Open":
+            rules = Rules(id=ns_tag,
+                          inbound_rules=[default_allow],
+                          outbound_rules=[default_allow])
+            _log.info("Applying Open Rules to NS Profile %s" % ns_tag)
+
+        elif self.policy == "Closed":
+            rules = Rules(id=ns_tag,
+                          inbound_rules=[Rule(action="allow", src_tag=ns_tag)],
+                          outbound_rules=[default_allow])
+            _log.info("Applying Closed Rules to NS Profile %s" % ns_tag)
+
+        else:
+            _log.error(
+                "Namespace %s policy is neither Open nor Closed" % self.name)
+            return False
+
+        # Write rules to profile
+        _datastore_client.etcd_client.write(
+            profile_path + "rules", rules.to_json())
+        return True
+
+    def delete_ns_profile(self):
+        # Derive NS tag
+        ns_tag = "namespace_%s" % self.name
```

Comment:
`ns_tag` is actaully the profile_name, correct?

As such, it should probably be called profile name instead.  It would probably make sense to create the name in `__init__` as self.profile_name so it can be referenced later.